src/snowflake/connector/auth.py

```diff
@@ -517,6 +571,34 @@ def write_temporary_credentials(self, host, user, session_parameters, response):
             )
         return
 
+    # Default timeout handler. This will trigger if the authenticator
+    # hasn't implemented one. By default we retry on timeouts and use
+    # jitter to deduce the time to sleep before retrying. The sleep
+    # time ranges between 1 and 16 seconds.
+    def handle_timeout(
+        self,
+        authenticator: str,
+        service_name: any,
```

Comment:
Shouldn't this be a `str`?
Also, I think [any](https://docs.python.org/3/library/functions.html#any) isn't what you meant anyway

Reply:
Later in this PR you declare this function as
```suggestion
        service_name: Optional[str],
```